vowpalwabbit/example_predict.h

```diff
@@ -12,7 +12,33 @@ typedef unsigned char namespace_index;
 #include "v_array.h"
 
 #include <vector>
+#include <set>
+#include <unordered_set>
 #include <array>
+// Mutex cannot be used in managed C++, tell the compiler that this is unmanaged even if included in a managed
+// project.
+#ifdef _M_CEE
+#  pragma managed(push, off)
+#  undef _M_CEE
+#  include <mutex>
+#  define _M_CEE 001
+#  pragma managed(pop)
+#else
+#  include <mutex>
+#endif
+
+struct namespace_interactions
```

Comment:
Not sure if we want to do it in this iteration, but is there a reason for not making new code more oop-ish?
It looks like it can be done as class without exposing all internals and expand_quadratics_wildcard_interactions as potential method.
I.e. size variable looks a bit confusing here - it is not clear which size it is actually + as far as I can see it is actually used as isUpdated flag which could be set to false on every add(namespace) call or something like that.

Reply:
There is no particular reason apart from the fact that, `interactions.interactions` vector will be used independently from everything else if we do not do `-q ::` and all the rest is only needed/used if we do `-q ::` and I opted for not disturbing the rest of the code if not

`size` does need a better name for sure and we could look into refactoring to make it more object oriented